From fd3b43caf46837a2095e02c9f95fc4dde86b035b Mon Sep 17 00:00:00 2001
From: Pierluigi Passaro <pierluigi.p@variscite.com>
Date: Sat, 3 Feb 2024 19:35:35 +0100
Subject: [PATCH 1/3] android-imx_device_fsl: allow building other NXP
 derivatives

These changes assume that the NXP derivatives are at the same level of
NXP device.
For Variscite SoMs this allows reusing most of the files provided by NXP
and just change what really required.

Signed-off-by: Pierluigi Passaro <pierluigi.p@variscite.com>
---
 common/build/Makefile    |  4 ++--
 common/tools/imx-make.sh | 11 ++++++-----
 2 files changed, 8 insertions(+), 7 deletions(-)

diff --git a/common/build/Makefile b/common/build/Makefile
index b4449a58..2be673c7 100644
--- a/common/build/Makefile
+++ b/common/build/Makefile
@@ -12,8 +12,8 @@ include ${soc_path}/UbootKernelCommonConfig.mk
 include ${product_path}/SharedBoardConfig.mk
 include ${product_path}/UbootKernelBoardConfig.mk
 
-include ${nxp_git_path}/common/build/kernel.mk
-include ${nxp_git_path}/common/build/uboot.mk
+include ${nxp_git_path}/../nxp/common/build/kernel.mk
+include ${nxp_git_path}/../nxp/common/build/uboot.mk
 -include ${product_path}/AndroidUboot.mk
 -include ${product_path}/AndroidTee.mk
 -include ${GPU_VIV6_PATH}/gpu-viv6/driver/hal/kernel/galcore.mk
diff --git a/common/tools/imx-make.sh b/common/tools/imx-make.sh
index 9fa70244..c91ae0e8 100755
--- a/common/tools/imx-make.sh
+++ b/common/tools/imx-make.sh
@@ -140,10 +140,11 @@ if [ -n "${build_kernel_modules}" ]; then
     build_kernel_oot_module_flag=1;
 fi
 
-product_makefile=`pwd`/`find device/nxp -maxdepth 4 -name "${TARGET_PRODUCT}.mk"`;
+product_makefile=`pwd`/`find device -maxdepth 4 -name "${TARGET_PRODUCT}.mk"`;
 product_path=${product_makefile%/*}
 soc_path=${product_path%/*}
 nxp_git_path=${soc_path%/*}
+export AARCH64_GCC_CROSS_COMPILE=$ANDROID_BUILD_TOP/prebuilts/gcc/linux-x86/aarch64/gcc-arm-8.3-2019.03-x86_64-aarch64-linux-gnu/bin/aarch64-linux-gnu-
 
 # if uboot is to be compiled, remove the UBOOT_COLLECTION directory
 if [ -n "${build_bootloader}" ]; then
@@ -152,7 +153,7 @@ fi
 
 # redirect standard input to /dev/null to avoid manually input in kernel configuration stage
 soc_path=${soc_path} product_path=${product_path} nxp_git_path=${nxp_git_path} clean_build=${clean_build} \
-    make -C ./ -f ${nxp_git_path}/common/build/Makefile ${parallel_option} \
+    make -C ./ -f ${nxp_git_path}/../nxp/common/build/Makefile ${parallel_option} \
     ${build_bootloader} ${build_kernel} </dev/null || exit
 # in the execution of this script, if the kernel build env is cleaned or configured, do not trigger that again
 if [ -n "${build_kernel}" ]; then
@@ -162,21 +163,21 @@ fi
 
 if [ -n "${build_kernel_modules}" ]; then
     soc_path=${soc_path} product_path=${product_path} nxp_git_path=${nxp_git_path} clean_build=${clean_build} \
-        skip_config_or_clean=${skip_config_or_clean} make -C ./ -f ${nxp_git_path}/common/build/Makefile ${parallel_option} \
+        skip_config_or_clean=${skip_config_or_clean} make -C ./ -f ${nxp_git_path}/../nxp/common/build/Makefile ${parallel_option} \
         ${build_kernel_modules} </dev/null || exit
     skip_config_or_clean=1
 fi
 
 if [ -n "${build_kernel_dts}" ]; then
     soc_path=${soc_path} product_path=${product_path} nxp_git_path=${nxp_git_path} clean_build=${clean_build} \
-        skip_config_or_clean=${skip_config_or_clean} make -C ./ -f ${nxp_git_path}/common/build/Makefile ${parallel_option} \
+        skip_config_or_clean=${skip_config_or_clean} make -C ./ -f ${nxp_git_path}/../nxp/common/build/Makefile ${parallel_option} \
         ${build_kernel_dts} </dev/null || exit
     skip_config_or_clean=1
 fi
 
 if [ ${build_kernel_oot_module_flag} -eq 1 ] || [ -n "${build_kernel_modules}" ]; then
     soc_path=${soc_path} product_path=${product_path} nxp_git_path=${nxp_git_path} clean_build=${clean_build} \
-        skip_config_or_clean=${skip_config_or_clean} make -C ./ -f ${nxp_git_path}/common/build/Makefile ${parallel_option} \
+        skip_config_or_clean=${skip_config_or_clean} make -C ./ -f ${nxp_git_path}/../nxp/common/build/Makefile ${parallel_option} \
         ${build_vvcam} ${build_galcore} ${build_mxmwifi} </dev/null || exit
 fi
 
-- 
2.34.1

